src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/temperature/CompactNodeSummary.java

```diff
@@ -68,6 +78,71 @@ public CompactNodeSummary(final String nodeId, final String hostAddress) {
         this.numOfShards = new int[TemperatureVector.Dimension.values().length];
     }
 
+    public static CompactNodeSummary buildSummaryFromDatabase(Result<Record> records,
+        DSLContext context) {
+        if (records.size() != 1) {
+            LOG.error("Expected 1 compact node summary, got {}. Summaries: {}", records.size(),
+                records);
+            throw new IllegalArgumentException(
+                "Only 1 CompactNodeSummary expected. Found: " + records.size());
+        }
+
+        Record record = records.get(0);
+        final String nodeId =
+            record.get(DSL.field(DSL.name(HotNodeSummary.SQL_SCHEMA_CONSTANTS.NODE_ID_COL_NAME),
```

Comment:
do we need to handle the DataTypeException here ?

Reply:
done.